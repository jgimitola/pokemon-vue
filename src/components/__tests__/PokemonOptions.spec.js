import { shallowMount } from "@vue/test-utils";
import { beforeEach, describe, expect, test } from "vitest";
import PokemonOptions from "@/components/PokemonOptions.vue";
import { pokemons } from "@/helpers/__tests__/mocks/pokemons.mock.js";

describe("PokemonOptions", () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons,
      },
    });
  });

  test("Match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Show options", () => {
    const items = wrapper.findAll("li");

    expect(items.length).toBe(4);

    const itemDisplayingNames = items.map((item) => item.text());
    const pokemonNames = pokemons.map((item) => item.name);

    expect(itemDisplayingNames).toStrictEqual(pokemonNames);
  });

  test("emit @selection", () => {
    const bulbasaurItem = wrapper.find("li");

    bulbasaurItem.trigger("click");

    expect(wrapper.emitted("selection").length).toBe(1);
    expect(wrapper.emitted("selection")[0]).toStrictEqual([pokemons[0]]);
  });
});
