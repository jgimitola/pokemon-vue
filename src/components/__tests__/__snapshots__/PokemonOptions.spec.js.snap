// Vitest Snapshot v1

exports[`PokemonOptions > Match snapshot 1`] = `
"<div data-v-d381d26c=\\"\\" class=\\"options-container\\">
  <ul data-v-d381d26c=\\"\\">
    <li data-v-d381d26c=\\"\\">bulbasaur</li>
    <li data-v-d381d26c=\\"\\">ivysaur</li>
    <li data-v-d381d26c=\\"\\">venusaur</li>
    <li data-v-d381d26c=\\"\\">charmander</li>
  </ul>
</div>"
`;
